upp/stages/normalisation.py

```diff
@@ -66,6 +66,9 @@ def get_class_dict(self, batch):
         for name, array in batch.items():
             if name != self.variables.jets_name:
                 array = array[array["valid"]]
+            if name == "jets":  # separate case for flavour_label
```

Comment:
I try to use `self.variables.jets_name` instead of hardcoding the string here. This makes things more general for non ftag (within ftag it rarely ever changes from `jets` but at some point it did in the past and it was a bit of a pain). We should also combine the check on `name` to check whether `flavour_label` exists and just skip adding it it doesn't for whatever reason

Reply:
Should the check for `flavour_label` be here or in lines [125-127](https://github.com/umami-hep/umami-preprocessing/pull/29/files#diff-e1a88c204340931eed9c60b4759e37d02c42f43cb6a94c140e753045093b35d6R126), where I add it to the list of variables for the `H5Reader`. If `flavour_label` doesn't exist, would `H5Reader just ignore it and read the rest, or would it cause problems?

Reply:
Still couldn't do that without opening the .h5 file. Hopefully this doesn't slow things down too much? 

See my implementation in [lines 127-130](https://github.com/umami-hep/umami-preprocessing/pull/29/files#diff-e1a88c204340931eed9c60b4759e37d02c42f43cb6a94c140e753045093b35d6R127-R130).

Reply:
Thanks! looks good. I still think we need a second check on the line here to check whether flavour_label is present, otherwise this looks good

Reply:
I'm sorry, I am not sure what you mean, doesn't `if "flavour_label" in f[self.jets_name].dtype.names:` do that already?

Reply:
Great thanks! I think we can remove the draft and merge this